pkg/events/derive/derive.go

```diff
@@ -2,14 +2,39 @@ package derive
 
 import (
 	"fmt"
+	"sync"
 
+	"github.com/aquasecurity/tracee/pkg/counter"
 	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/events/parse"
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
 // Use as static variable for testability reasons
 var getEventDefFunc = events.Definitions.Get
 
+// Context is a struct used to store the triggering events' context
+type Context struct {
+	Map     map[uint64]trace.Event
+	Mutex   sync.Mutex
+	Counter counter.Counter
+}
+
+func (*Context) AddToMap(event trace.Event) uint64 {
+	// Initial event - no need to save
+	if event.Timestamp == 0 {
+		return 0
+	}
+	contextMapID := uint64(InvokedContext.Counter.Read())
+	InvokedContext.Counter.Increment(1)
+	InvokedContext.Mutex.Lock()
+	InvokedContext.Map[contextMapID] = event
+	InvokedContext.Mutex.Unlock()
+	return contextMapID
+}
+
+var InvokedContext = Context{make(map[uint64]trace.Event), sync.Mutex{}, counter.Counter(1)}
```

Comment:
As a compromise, Instead of exporting this global instance (basically a singleton), I think we can have an upper `AddToMap` which is exported instead of the whole struct. This is then shared internally with the module.

However, as mentioned, the best solution would be to not have shared state though, this should get it's own module and injected from tracee into the deriving module. Then we can easily add tests for this mechanism, which are currently missing.

Reply:
As we will change this mechanism in the next version, is that really important that we change this and only export the `AddToMap` function? we will change it anyway once we create the `triggeredEvents` package.

Reply:
Global exported variables are just a problem waiting to happen a lot of the times. On that chance I don't want to leave it like this before merging.